src/geocel/Types.hh

```diff
@@ -77,6 +78,7 @@ struct GeoTrackInitializer
 {
     Real3 pos{0, 0, 0};
     Real3 dir{0, 0, 0};
+    TrackSlotId parent;
```

Comment:
Given how this is being used (i.e., it's not actually used by `operator=(Initializer_t const&)`, but rather used by the core track to construct `AdvancedInitializer const&`) I don't think it belongs here. (Unless the usage in core track view is "temporary" in which case it might need a to-do. I think a change that would "fix" this problem is to remove the "advanced initializer" nested structs entirely and instead have the GeoTrackInitializer directly pull from the other state. The downside is that would require keeping a pointer to the whole state vector rather than just the local data.

Reply:
Right, my understanding was we're hoping to eventually replace this optimization with a fast initialization of the geometry state, and whatever information we need for that would live here. I'll try removing the detailed initializers and see how that looks.

Reply:
Does that seem any better? If not I can revert it and just move the parent ID up a level in the track initializer.